web/src/composebox_typeahead.ts

```diff
@@ -1091,6 +1094,22 @@ export function content_typeahead_selected(
             message: "",
         };
     }
+
+    // We only want to consider escape key for the stream+topic typeahead completion case.
+    if (event?.key === "Escape" && item.type !== "topic_list") {
+        setTimeout(() => {
+            // Select any placeholder text configured to be highlighted.
+            if (highlight.start && highlight.end) {
+                $textbox.range(highlight.start, highlight.end);
+            } else {
+                $textbox.caret(beginning.length);
+            }
+            // Also, trigger autosize to check if compose box needs to be resized.
+            compose_ui.autosize_textarea($textbox);
+        }, 0);
+        return beginning + rest;
+    }
```

Comment:
Why is `setTimeout` needed here? That should be documented in the comments.

I also feel like this section is missing framing for what this block of code is about -- the comment at the start is just repeating what the conditional does, but what someone reading this part of the file will want to know is what scenario this code path is about handling, so they can reason about it.

Reply:
Why are you copying this code here? Could you not just do

```js
if (event?.key === "Escape" && item.type !== "topic_list") {
   break;
}
```

Then the code you copied below would run.

Reply:
Looks like the timeout was added originally in 2013 [here](https://github.com/zulip/zulip/commit/771eea22b8d760c9c38d1e8e9fb32d0089e8d7a4#diff-dbf09ea24455151b484cb21bff38a79e26c69887dd89a0a18830f29292998257R219)

Reply:
Oh oops I thought this was inside the switch statement for some reason. I see why you did this now.

> Should I extract the last part into a function and call it twice?

I do think that would be better, yeah.